src/ui/components/PasswordModule/PasswordModule.tsx

```diff
@@ -44,25 +49,43 @@ const PasswordModule = forwardRef<PasswordModuleRef, PasswordModuleProps>(
       setHintValue("");
     };
 
+    const handleClearExisting = () => {
+      setAlertExistingIsOpen(false);
+      handleClearState();
+    };
+
     useImperativeHandle(ref, () => ({
       clearState: handleClearState,
     }));
 
     const handleContinue = async (skipped: boolean) => {
       if (!skipped) {
-        await SecureStorage.set(
-          KeyStoreKeys.APP_OP_PASSWORD,
-          createPasswordValue
+        const currentPassword = await SecureStorage.get(
+          KeyStoreKeys.APP_OP_PASSWORD
         );
-        if (hintValue) {
-          await Agent.agent.basicStorage.createOrUpdateBasicRecord(
-            new BasicRecord({
-              id: MiscRecordId.OP_PASS_HINT,
-              content: { value: hintValue },
-            })
+        if (
+          isModal &&
+          authentication.passcodeIsSet &&
+          currentPassword === createPasswordValue
+        ) {
+          setAlertExistingIsOpen(true);
+          return;
+        } else {
+          await SecureStorage.set(
+            KeyStoreKeys.APP_OP_PASSWORD,
+            createPasswordValue
           );
+          if (hintValue) {
+            await Agent.agent.basicStorage.createOrUpdateBasicRecord(
+              new BasicRecord({
+                id: MiscRecordId.OP_PASS_HINT,
+                content: { value: hintValue },
+              })
+            );
+          }
         }
       } else {
+        await SecureStorage.set(KeyStoreKeys.APP_OP_PASSWORD, "");
```

Comment:
Hmm, why are we storing empty values in the secure storage? We already have `APP_PASSWORD_SKIPPED`

Reply:
see second last comment - `.delete`

Reply:
There was an edge case we found when we first built the password feature that required both in order to know if the user was there for the first time, that's why myself and Jaime decided to have both. I will try to change it following your suggestions in the other comment and see how it goes if we can get rid of it.

Reply:
First time should be that there's nothing in the secure storage, and `APP_PASSWORD_SKIPPED` is also not set

Reply:
yep thats all good